lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -158,7 +165,7 @@ def firewall_command(cmdlet_type)
           cmd << " -Direction '#{new_resource.direction}'" if new_resource.direction
           cmd << " -Protocol '#{new_resource.protocol}'" if new_resource.protocol
           cmd << " -Action '#{new_resource.firewall_action}'" if new_resource.firewall_action
-          cmd << " -Profile '#{new_resource.profile}'" if new_resource.profile
+          cmd << " -Profile '#{new_resource.profile.join("', '")}'" if new_resource.profile
```

Comment:
I have not used this command in anger, so I looked up [the documentation for New-NetFirewallRule](https://docs.microsoft.com/en-us/powershell/module/netsecurity/new-netfirewallrule?view=win10-ps#parameters). It states for the `-Profile` parameter:

> Separate multiple entries with a comma and do not include any spaces.

Have you seen this change work on a Windows host? I see that the unit test was updated (and so passes) to check that the generated command includes `-Profile 'private', 'public'`, but will that space cause an error at run time?

Reply:
Didn't get that far yet, but it appears you're right. If they've written the cmdlet like that it's definitely not idiomatic PowerShell... I was about to say that this explains why LocalPort and RemotePort are written that way in the resource, but those parameters are actually documented as taking `String[]`

Reply:
So turns out both approaches work for `Profile`, whereas the comma delimited string the resource uses for `LocalPort` and `RemotePort` actually doesn't work.

Reply:
Well, thanks for finding that out and fixing them!